### DISCUSSION TITLE
License AGPLv3

### DISCUSSION THREAD

**Discussion by Gellish** [givanz/Vvveb#135] [Q&A] [answered]
Hello! I have a question about theme creation on a CMS with the AGPLv3 license. I understand it's a complex topic, but I'm trying to figure out if the theme I create on this CMS will consist solely of my own code or if it will include elements subject to the AGPLv3. Any insights or guidance would be greatly appreciated. Thanks in advance!

**Comment by givanz** [comment, marked as answer]
Hi

Themes can have any license, most themes have MIT license https://github.com/topics/vvveb-theme the default Landing theme has APLv2 https://github.com/givanz/landing?tab=Apache-2.0-1-ov-file#readme

They don't include code from the CMS directly, they only link to the default js library for ajax calls such as app.js https://github.com/givanz/landing/blob/main/src/_includes/footer-scripts.html#L21 but the code is not included.

No AGPL code is included in themes code.

You can use the default theme or any other theme as a reference or base to build your own.

**Reply by goshawk1980** [reply]
Hi givanz,

In addition to the question of Gellish :

First I have to say this is awesome work you've done! and quiet similar to something i am building right now, lol :)
So much I seriously consider to stop doing so and convert my "missing features" into a Vvveb plugin instead.

If I would do that, do plugins fall under that AGPLv3 license -or- are these just like templates and can they have any license?
The latter seems more logical as they are not part of the core code but i preferred to ask.

**Reply by givanz** [reply]
Hi

Thank you.
Plugins like themes can be released with any license.

**Comment by Gellish** [comment]
when your "Vvveb" plugin is licensed under AGPLv3, any plugins you create for that **must also be licensed in an AGPLv3** 
It’s recommended to use licenses like MIT for "Vvveb" plugin because example your work "missing features had AGPLv3 is okay because it’s compatible with both AGPLv3 and commercial licenses

Mean while your Vvveb theme I don't see is license or maybe is part core?

**Comment by givanz** [comment]
> It’s recommended to use licenses like MIT for "Vvveb" plugin because example your work "missing features had AGPLv3 is okay because it’s compatible with both AGPLv3 and commercial licenses

All plugins are released under GPLv3 https://github.com/topics/vvveb-plugin but plugins can be released with any license.

> Mean while your Vvveb theme I don't see is license or maybe is part core?

Both themes that are included in the default installation package are APLv2 https://github.com/givanz/landing/blob/main/LICENSE https://github.com/givanz/blog-default/blob/main/LICENSE

**Reply by Gellish** [reply]
Great!
